ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -53,11 +53,23 @@ public DDLCommandResult run(MetaStore metaStore) {
           dataSourceType == DataSource.DataSourceType.KSTREAM ? "STREAM" : "TABLE"
       ));
     }
-    DropTopicCommand dropTopicCommand =
-        new DropTopicCommand(dataSource.getKsqlTopic().getTopicName());
+    if (!metaStore.isSafeToDrop(sourceName)) {
```

Comment:
This looks the same as in `MetaStoreImpl#deleteSource` should only have the logic in one place